src/ObjectMakr.ts

```diff
@@ -73,7 +72,9 @@ export class ObjectMakr implements IObjectMakr {
             this.processProperties(this.properties);
         }
 
-        this.processFunctions(this.inheritance, Object, "Object");
+        if(!this.functions){
```

Comment:
for the little bit here in the processFunctions for ObjectMakr.ts:
``` javascript
for (let ref in this.properties[name]) {
    if (this.properties[name].hasOwnProperty(ref)) {
        this.functions[name].prototype[ref] = this.properties[name][ref];
    }
}
```
do you want the properties addition to be in its own check...right now I have it grouped with a check to see if `this.functions[name]` needs to be initialized